src/System.Private.CoreLib/src/Internal/Reflection/Core/NonPortable/RuntimeTypeUnifier.cs

```diff
@@ -65,51 +67,58 @@ private sealed class RuntimeTypeHandleToTypeCache : ConcurrentUnifierW<IntPtr, T
             protected sealed override Type Factory(IntPtr rawRuntimeTypeHandleKey)
             {
                 EETypePtr eeType = new EETypePtr(rawRuntimeTypeHandleKey);
-                RuntimeTypeHandle runtimeTypeHandle = new RuntimeTypeHandle(eeType);
+                return GetRuntimeTypeBypassCache(eeType);
+            }
 
-                // Desktop compat: Allows Type.GetTypeFromHandle(default(RuntimeTypeHandle)) to map to null.
-                if (runtimeTypeHandle.IsNull)
-                    return null;
+            public static readonly RuntimeTypeHandleToTypeCache Table = new RuntimeTypeHandleToTypeCache();
+        }
 
-                ReflectionExecutionDomainCallbacks callbacks = RuntimeAugments.Callbacks;
+        // This bypasses the CoreLib's unifier, but there's another unifier deeper within the reflection stack:
+        // this code is safe to call without locking. See comment above.
+        public static Type GetRuntimeTypeBypassCache(EETypePtr eeType)
+        {
+            RuntimeTypeHandle runtimeTypeHandle = new RuntimeTypeHandle(eeType);
 
-                if (eeType.IsDefType)
-                {
-                    if (eeType.IsGenericTypeDefinition)
-                    {
-                        return callbacks.GetNamedTypeForHandle(runtimeTypeHandle, isGenericTypeDefinition: true);
-                    }
-                    else if (eeType.IsGeneric)
-                    {
-                        return callbacks.GetConstructedGenericTypeForHandle(runtimeTypeHandle);
-                    }
-                    else
-                    {
-                        return callbacks.GetNamedTypeForHandle(runtimeTypeHandle, isGenericTypeDefinition: false);
-                    }
-                }
-                else if (eeType.IsArray)
-                {
-                    if (!eeType.IsSzArray)
-                        return callbacks.GetMdArrayTypeForHandle(runtimeTypeHandle, eeType.ArrayRank);
-                    else
-                        return callbacks.GetArrayTypeForHandle(runtimeTypeHandle);
-                }
-                else if (eeType.IsPointer)
+            // Desktop compat: Allows Type.GetTypeFromHandle(default(RuntimeTypeHandle)) to map to null.
```

Comment:
Looks like this was moved to `Type.GetTypeFromHandle` and it should not be needed here anymore